crates/uv/src/commands/project/environment.rs

```diff
@@ -71,7 +71,9 @@ impl CachedEnvironment {
 
         // Hash the interpreter based on its path.
         // TODO(charlie): Come up with a robust hash for the interpreter.
-        let interpreter_hash = cache_digest(&interpreter.sys_executable());
+        let interpreter_hash = cache_digest(&uv_fs::canonicalize_executable(
+            interpreter.sys_executable(),
+        )?);
```

Comment:
Is this safe? Can't the interpreter path change other behaviors?

Reply:
I think we explicitly avoid canonicalizing elsewhere, e.g., `Interpreter::query_cached`. I think you may need to use your symlink directory detection to only canonicalize when safe.